typescript/trans/typechecking/typescript.nabl2

```diff
@@ -146,8 +161,26 @@ rules
       Value{name} : ty !,
       [[ type ^ (s) : ty ]].
     
-    [[ ValueReturn(value, _) ^ (s) : ty ]] :=
-      [[ value ^ (s) : ty ]].
+    [[ If(condition, trueBlock) ^ (s) ]] :=
+      [[ condition ^ (s) : valueTy ]],
+      valueTy == BOOLEAN() | error $[Boolean expected, got [valueTy]],
+      [[ trueBlock ^ (s) ]].
+    
+    [[ IfElse(condition, trueBlock, falseBlock) ^ (s) ]] :=
+      [[ condition ^ (s) : valueTy ]],
+      valueTy == BOOLEAN() | error $[Boolean expected, got [valueTy]],
+      [[ trueBlock ^ (s) ]],
+      [[ falseBlock ^ (s) ]].
+    
+    [[ Block(statements) ^ (s) ]] :=
+      Map1 [[ statements ^ (s) ]].
+    
+    [[ ValueReturn(value, _) ^ (s) ]] :=
+      [[ value ^ (s) : valueTy ]],
+      Return{"function"} -> s,
+      Return{"function"} |-> d | error $[Can not use return outside of a function scope],
+      d: ty,
+      ty == valueTy.
```

Comment:
So this actually cant be `<?`, because the variable would then be unbound. Not sure how we can say: 'If this right thing is not known, just set it to my value'.